### PR TITLE
ci: add better version check github action

### PR DESCRIPTION
**Pull request by a-dubs**
Improved the github CI/CD version check so that it not only ensures the version has been bumped, but that the version bump is sane. 

The following tests were run against the version check to ensure it functions as desired:
```python
class TestVersionComparison(unittest.TestCase):
    def test_pass_cases(self):
        self.assertEqual(compare_versions("1!1.0.0", "1!1.0.1"), "Version bump is sane.")
        self.assertEqual(compare_versions("1!1.0.0", "1!1.1.0"), "Version bump is sane.")
        self.assertEqual(compare_versions("1!1.1.1", "1!2.0.0"), "Version bump is sane.")
        self.assertEqual(compare_versions("1!1.1.0", "1!1.1.1"), "Version bump is sane.")
        self.assertEqual(compare_versions("1!1.1.1", "1!1.1.2"), "Version bump is sane.")

    def test_fail_cases(self):
        self.assertNotEqual(compare_versions("1!1.0.0", "1!1.0.0"), "Version bump is sane.")
        self.assertNotEqual(compare_versions("1!1.0.0", "1!1.2.1"), "Version bump is sane.")
        self.assertNotEqual(compare_versions("1!1.0.0", "1!2.1.0"), "Version bump is sane.")
        self.assertNotEqual(compare_versions("1!1.0.0", "1!0.9.9"), "Version bump is sane.")
        self.assertNotEqual(compare_versions("1!1.1.2", "1!1.1.1"), "Version bump is sane.")
        self.assertNotEqual(compare_versions("1!1.2.1", "1!1.1.1"), "Version bump is sane.")
        self.assertNotEqual(compare_versions("1!2.1.1", "1!1.1.1"), "Version bump is sane.")
```

I also already opened a temporary PR against my own fork to ensure the github action behaved as expected.

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1!6.2.2
+1!6.2.3
```
Comment (a-dubs) [inline VERSION:1]:
```suggestion
1!6.2.4
```

File: .github/workflows/version_check.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    return "Version bump is sane."
+
+if __name__ == "__main__":
+    if len(sys.argv) != 3:
+        print("Usage: python version_check.py <old_version> <new_version>")
+        sys.exit(1)
+    
+    result = compare_versions(sys.argv[1], sys.argv[2])
+    print(result)
+    if result != "Version bump is sane.":
```
Comment (blackboxsw) [inline .github/workflows/version_check.py:32]: Generic inclusive language nit.
```suggestion
    if not result.startswith("Valid version"):
```

**Review by blackboxsw — COMMENTED** [review]
Generally looks good.
Minimally the following should be resolved:
- I'd change the success message to include Valid version instead of what we have.
- isort and black formatting issues with the added scrip

Optionally:
- You could consider using packaging.version for this comparison for readability, but it's your call as it's about the same amount of code.

File: .github/workflows/version_check.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+def compare_versions(old_version, new_version):
+    if old_version >= new_version:
+        return f"Version {old_version} must be incremented."
+    if not re.match(r"\d!\d+.\d+.\d+", new_version.public):
+        return f"Version requires <epoch>!<major>.<minor>.<patch>. Found: {new_version}"
+    major_release = new_version.major > old_version.major
+    minor_release = new_version.minor > old_version.minor
+    micro_release = new_version.micro > old_version.micro
+    if sum([major_release, minor_release, micro_release]) > 1:
+        return "Only one digit can be incremented at a time.
```
Comment (holmanb) [inline .github/workflows/version_check.py:12]: invalid python, missing double quote

**Review by holmanb — CHANGES_REQUESTED** [review]

File: .github/workflows/version_check.py (hunk 3 above)
Comment (a-dubs) [reply .github/workflows/version_check.py:12]:
```suggestion
        return "Only one digit can be incremented at a time."
```

**Comment by a-dubs** [comment]
dear god, whoever merges this, please squash this heretical commit history :skull: :skull: :skull:

**Comment by blackboxsw** [comment]
> dear god, whoever merges this, please squash this blasphemous commit history 💀 💀 💀

@a-dubs you can squash merge at will :)
```
git reset --sort HEAD~10
git commit # add your desired commit message
```

**Review by holmanb — APPROVED** [review]
thanks @a-dubs!

**Pull request merged**